seacrowd/sea_datasets/seaeval/seaeval.py

```diff
@@ -0,0 +1,238 @@
+                gen_kwargs={
+                    "paths": paths,
+                    "split": "train",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, paths: Path, split: str) -> Tuple[int, Dict]:
+        """
+        Yields examples as (key, example) tuples.
+        """
+
+        language = self.config.subset_id.split("_")[3]
+        examples = None
+
+        for key, path in paths.items():
+            if "cross" in key:
+                data = pd.read_json(path).rename(columns=self.LANGUAGES_EXCHANGED)
+                data = pd.melt(data, id_vars=["id"], value_vars=_LANGUAGES.keys(), var_name="language")
+                data_flattened = pd.json_normalize(data["value"])
+                data_merged = pd.merge(data, data_flattened, left_index=True, right_index=True)
+                data_filtered = data_merged[data_merged["language"] == language].drop(columns=["value", "language"])
+                examples = data_filtered.to_records()
+            elif "eval" in key:
+                data = pd.read_json(path)
+                examples = data.to_records()
+
+        idx = 0
+        if self.config.schema == "source" and self.config.subset_id not in ["cross_logiqa", "ph_eval"]:
+            for row in examples:
+                x = {
+                    "id": row["id"],
+                    "question": row["question"],
+                    "choices": row["choices"],
+                    "answer": row["answer"],
+                }
+                yield idx, x
+                idx += 1
+        elif self.config.schema == "source" and self.config.subset_id == "cross_logiqa":
+            for row in examples:
+                x = {
+                    "id": row["id"],
+                    "question": row["question"],
+                    "context": row["context"] if "context" in row else None,
+                    "choices": row["choices"],
+                    "answer": row["answer"],
+                }
+                yield idx, x
+                idx += 1
+        elif self.config.schema == "source" and self.config.subset_id == "ph_eval":
+            for row in examples:
+                x = {
+                    "id": row["id"],
+                    "question": row["question"],
+                    "choices": row["choices"],
+                    "answer": row["answer"],
+                    "category": row["category"] if "category" in row else None,
+                }
+                yield idx, x
+                idx += 1
+        elif self.config.schema == "seacrowd_qa":
+            for row in examples:
+                x = {
+                    "id": idx,
+                    "question_id": row["id"],
+                    "document_id": row["id"],
+                    "question": row["question"],
+                    "type": "multiple_choice",
+                    "choices": row["choices"],
+                    "context": row["context"] if "context" in row else None,
```

Comment:
How about:

```
row.get("context")
```

For this type of logic? It returns `None` if the key itself doesn't exist.

Reply:
The `row` object is of type `numpy.recarray` and doesn't have a `.get()` method. There doesn't seem to be a built-in method for `nump.recarray` that has the same behavior as `.get()` for Python `dict` types.